xonsh/built_ins.py

```diff
@@ -180,7 +182,9 @@ def expand_path(s):
     global ENV
     if ENV is not None:
         ENV.replace_env()
-    return os.path.expanduser(os.path.expandvars(s))
+    if ENV.get('EXPAND_ENV_VARS'):
+        s = expandvars(s)
```

Comment:
This still calls `ENV.replace_env()`, which now seems either unnecessary or totally sufficient.